infrastructure/templates/public-api/main.bicepparam

```diff
@@ -0,0 +1,116 @@
+using './main.bicep'
```

Comment:
the stuff in this file **ALMOST** (but not quite) looks like a duplicate of the stuff at the top of `main.bicep`. Do we need both?

Reply:
I'm not entirely sure what this file is used for so this is quite possibly just a lack of knowledge on my part (apologies if so!)

Reply:
Its a useful tool to have, I have kept most items as default, but I wanted to include the ability to make changes to the parameter file whilst leaving the main.bicep untouched. It is also best practice (MS)

Reply:
I think this was simply me not understanding how these files were used.

Is there a reason it doesn't match the ones in `main.bicep` exactly though?

Reply:
Jack's correct, this doesn't look right at all.

I think this should only really contain parameters that are overriding parameters in  `main.bicep`. Parameters that are using the default shouldn't need to be re-declared here!

We've done this with the current infrastructure where we have different `.parameters.json` files for each respective environment. We then deploy the infrastructure for each environment by specifying the parameter file to use.

We should do a similar thing here, but using `bicepparam` files instead. We can do a similar thing to the current infrastructure and create a new `parameters` directory under `public-api` that contains these files.